crates/bevy_ui/src/layout/mod.rs

```diff
@@ -161,12 +172,17 @@ without UI components as a child of an entity with UI components, results may be
         &mut self,
         parent_window: Entity,
         children: impl Iterator<Item = Entity>,
-    ) {
+    ) -> bool {
         let taffy_node = self.window_nodes.get(&parent_window).unwrap();
         let child_nodes = children
             .map(|e| *self.entity_to_taffy.get(&e).unwrap())
             .collect::<Vec<taffy::node::Node>>();
-        self.taffy.set_children(*taffy_node, &child_nodes).unwrap();
+        if self.taffy.children(*taffy_node).unwrap() != child_nodes {
+            self.taffy.set_children(*taffy_node, &child_nodes).unwrap();
```

Comment:
Why not bake that check into Taffy's `set_children()` directly and have Taffy itself tell us if any change occurred? That sounds better conceptually, and allows Taffy internally to optimize any `Vec` comparison if possible (can maybe run the `Vec` once looking for changes and applying them in one pass?).

Reply:
> Why not bake that check into Taffy's `set_children()` directly and have Taffy itself tell us if any change occurred? That sounds better conceptually, and allows Taffy internally to optimize any `Vec` comparison if possible (can maybe run the `Vec` once looking for changes and applying them in one pass?).

Probably what we'll do in the future is have our own implementation of  Taffy's `LayoutTree` rather than using the `Taffy` built-in. With a custom `LayoutTree` impl the Taffy algorithm could access Bevy's parent-child tree directly and there would be no need for a second internal tree with all the fragile synchronization logic that entails.

Reply:
Ok thanks for the details!